cdcv2/metadata/sql/client.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import (
+	"context"
+	"database/sql"
+	"time"
+
+	"gorm.io/gorm"
+)
+
+// TxnAction is a series of operations that can be executed in a transaction and the
+// generic type T represents the transaction context.
+//
+// Note that in the current implementation the metadata operation and leader check are
+// always in the same transaction context. In the future, cross-database transactions
+// could also be supported with different implementations.
```

Comment:
What does `cross-database transactions` mean? Why we need this?

Reply:
The comments here are somewhat ambiguous. In fact, there is no need for cross-database transactions in the current implementation. And i will remove such comments.